Purchases/Purchasing/StoreKitAbstractions/SK1StoreTransaction.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SK1StoreTransaction.swift
+//
+//  Created by Nacho Soto on 1/4/22.
+
+import StoreKit
+
+internal struct SK1StoreTransaction: StoreTransactionType {
+
+    init(sk1Transaction: SK1Transaction) {
+        self.productIdentifier = sk1Transaction.productIdentifier ?? ""
+        self.purchaseDate = sk1Transaction.purchaseDate
+        self.transactionIdentifier = sk1Transaction.transactionID
+    }
+
+    let productIdentifier: String
+    let purchaseDate: Date
+    let transactionIdentifier: String
+
+}
+
+extension SKPaymentTransaction {
+
+    /// Considering issue https://github.com/RevenueCat/purchases-ios/issues/279, sometimes `payment`
+    /// and `productIdentifier` can be `nil`, in this case, they must be treated as nullable.
+    /// Due of that an optional reference is created so that the compiler would allow us to check for nullability.
+    var productIdentifier: String? {
+        guard let payment = self.payment as SKPayment? else {
+            Logger.appleWarning(Strings.purchase.skpayment_missing_from_skpaymenttransaction)
+            return nil
+        }
+
+        guard let productIdentifier = payment.productIdentifier as String?,
+              !productIdentifier.isEmpty else {
+                  Logger.appleWarning(Strings.purchase.skpayment_missing_product_identifier)
+                  return nil
+              }
+
+        return productIdentifier
+    }
+
+    fileprivate var purchaseDate: Date {
+        guard let date = self.transactionDate else {
+            Logger.appleWarning(Strings.purchase.sktransaction_missing_transaction_date)
+
+            return Date()
```

Comment:
I don't love this, but it was the only solution I could think of. Making the values `nullable` just for wrapping `SK1Transaction`s makes everything worse.

Reply:
Maybe we should create a single date object that is a known bad date. Like Jan 1 1970, and always return that?

Reply:
I like that!